### ISSUE TITLE
YOLOv8-l-seg freezing on training on random epoch

### ISSUE THREAD

**Issue by Irikos** [ultralytics/ultralytics#1621]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

I'm training YOLOv8 for segmentation on a custom dataset. Started with n, s, m and now, at l, it simply freezes at some random epoch. I tried it many times, it doesn't always freeze on the same epoch etc. Any idea why this could be happening? It is worth noting that the smaller models had no problem in training on this dataset and with similar - or identical - parameters.

In training, I end up using 17.8GB out of total 24GB that I have, so I assume this is not the fault (I even increased the batch size a bit and it froze later).

Most of the time it froze around epoch 100 out of 300 (93, 103, 107 etc)

### Environment

Ultralytics YOLOv8.0.53 🚀 Python-3.10.4 torch-1.12.1 CUDA:0 (NVIDIA GeForce RTX 3090, 24246MiB)

### Minimal Reproducible Example

!yolo task=segment mode=train model=yolov8l-seg.pt data=$current_yaml epochs=300 imgsz=640 batch=24 workers=24 name=$current_iteration

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Irikos you're likely just running out of resources since it doesn't occur with smaller models. Use a smaller batch sizes or smaller image size. 

I'm going to attach a non-reproducible tag here since by your description it's hard to reproduce. If you can produce a reproducible example that we can see ourselves, i.e. COCO128 in Colab let us know.

**Comment by Irikos** [comment]
I understand and agree to the non-reproducible tag; unfortunately, I cannot provide a colab version now, as the dataset is not (yet) public and I haven't had this problem on coco. Once I make the dataset public, I will try to provide a colab as well.

I don't know why the I would be running out of resources, though, as it froze more when the batch size was smaller. Increasing the batch size yielded less such situations (on identical dataset split).

**Comment by glenn-jocher** [comment]
@Irikos got it, thank you! Yes please let us know when a reproducible example is available and we will take a look.

**Comment by nikolaydyankov** [comment]
I'm experiencing the same issue and Google brought me to this issue. The training process will freeze randomly. I'm running the same experiment with over 1GB VRAM to spare and sometimes it will finish, other times it will just hang at a random epoch.

How can I begin to troubleshoot this?

**Comment by glenn-jocher** [comment]
@nikolaydyankov, to troubleshoot this issue, first, confirm whether other deep learning frameworks have similar issues with out-of-memory under the same conditions. If other frameworks are also facing the same issue, it's likely an overfitting problem, and you may need to perform regularization. If other frameworks work reliably, it could suggest a bug in the YOLOv8 code. 

You might also try reducing both the batch size and image size, even if there is spare VRAM, as this may help solve the issue. If the problem persists, following the "Tips for Best Training Results" guide under the "Training" portion of the [YOLOv8 Docs](https://docs.ultralytics.com/), examining the model's performance on the validation set, increasing regularization, and looking for signs of overfitting may help solve the issue.

**Comment by nikolaydyankov** [comment]
I want to figure out what went wrong, not change hyperparameters or try a different framework. I'm sure it's not a VRAM issue, because it happens randomly on lower batch sizes as well with over 10GB vram to spare.

**Comment by glenn-jocher** [comment]
Understood @nikolaydyankov - to further diagnose the problem, I would recommend examining the training logs to see whether the freeze happens during backpropagation, during a function call or somewhere else. If the code is frozen, you can interrupt the kernel and use a debugger to identify where the code hangs. This will help you identify why the training process is freezing randomly. 

Another suggestion is to start training from an earlier checkpoint and resuming training, to see whether that is able to overcome the issue. This can help narrow down whether the problem lies in the implementation or the entire system. Once the issue has been identified, you can submit a bug report on the Ultralytics YOLOv8 GitHub Issues page with this information, and we will try to help you come up with a solution if possible.

**Comment by nikolaydyankov** [comment]
Thank you, I will get to it.

**Comment by glenn-jocher** [comment]
Absolutely, @nikolaydyankov! To diagnose your issue, I recommend examining the training logs to see where the freeze happens. Interrupting the kernel and using a debugger can help you identify where the code hangs. Starting training from an earlier checkpoint and resuming training can help know whether the issue lies in the entire system or it's specific to the current situation. Identifying the problem can help you submit a bug report on the Ultralytics YOLOv8 GitHub Issues page. Our team will be happy to help you solve the issue if possible.

**Comment by mzahana** [comment]
@Irikos @nikolaydyankov  I am facing similar issue. Did you find a solution.
I am using CUDA `12.2`, pytorch `2.1.2+cu121 and Nvidia RTX 4070

**Comment by Irikos** [comment]
Hey @mzahana! Unfortunately, no. I simply restarted several times and it worked (it did not always stop in the same place). It so happens that I am training on the same dataset again these days and I am curious if the problem will arise again.

**Comment by tejasri19** [comment]
Did you solve the problem? @Irikos . Interested to know the solution

**Comment by Irikos** [comment]
@tejasri19 not actively, unfortunately. At that time I just started it multiple times until it did not crash anymore. And I switched to a 4090 in the meantime and a fresh install of Ubuntu. Not sure if it is connected or not, but I stopped having that problem so I couldn't properly debug it anymore.
